### PR TITLE
fix(scap/gvisor): do not use C99 initializers in C++ in gVisor

### PR DESCRIPTION
**Pull request by LucaGuerra**
Signed-off-by: Luca Guerra <luca@guerra.sh>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

/area libscap-engine-gvisor

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

After seeing this https://github.com/falcosecurity/libs/pull/511 I noticed that C99-style initializers are only officially supported in C++ starting from C++20. In practice, it may work in some compilers (such as my own and the CI's one) but error in others. Now, this makes me sad because if fields change in the scap structure definition it'll be harder to update, but I don't think there is any way to make it better in C++ at this point.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

cc @gnosek could you please check if it works on your compiler?

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Andreagit97 — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/libscap/engine/gvisor/gvisor.cpp [hunk 1]
```diff
@@ -150,27 +150,27 @@ static int32_t gvisor_getpid_global(struct scap_engine_handle engine, int64_t* p
 #endif
 
 extern const struct scap_vtable scap_gvisor_engine = {
```
Comment (FedeDP) [inline userspace/libscap/engine/gvisor/gvisor.cpp:152]: This is so bad for readability though...

**Pull request merged**